### ISSUE TITLE
Nobody is receiving items - OP or not OP.

### ISSUE THREAD

**Issue by chumpydo** [RockinChaos/ItemJoin#116]
Hey there,

I am testing the plugin and am using the example item included in the configuration, see here:
` book-item:
    id: WRITTEN_BOOK
    slot: 8
    name: '&c&nCreating a Book'
    author: '&bItemJoin'
    lore:
      - '&7'
      - '&7This magical book is given on join!'
    pages:
      - 'newpage: &bYou can get this book on join!'
      - 'newline: &aThis book also supports &ccolor codes!'
      - 'newline: &0'
      - 'newline: &cHow do custom books work?'
      - 'newline: &0'
      - 'newline: &0You can set an &cauthor&0'
      - 'newline: &0as shown in the items.yml'
      - 'newline: &0I set it to &bItemJoin.'
      - 'newline: &0'
      - 'newpage: &cHow do you create a new page?'
      - 'newline: &0'
      - 'newline: &0As displayed to the left in the items.yml'
      - 'newline: &0Use &bnewpage&0 and &cnewline&0 to create your pages.'
      - 'newline: &cnewpage &0will start the next page.'
      - 'newline: &cnewline &0will create a new line for that page.'
      - 'newpage: &bItemJoin &0by RockinChaos!'
      - ':endthebook:'
    itemflags: inventory-modify, death-drops, self-drops
    triggers: join, respawn
    permission-node: 'itemjoin.book'
    enabled-worlds: world`
However, this is not working. I have two accounts, one is an OP, one is not. Nobody receives the item on join. Help?

Best,
Nathan Sykes // Chump

**Comment by RockinChaos** [comment]
I cant really tell what that is, can you post your full items.yml in a pastebin.com and send it to me.

Its one of two things, either your defining the wrong world or you are using the old config structure after the .yml reset.

**Comment by chumpydo** [comment]
Yep, here you are: https://hastebin.com/okegixahas.php

**Comment by RockinChaos** [comment]
Yeah your missing the items Identifier theres just a huge gab inbetween defining the item. One sec ill fix it and send a pastebin.com link.

**Comment by chumpydo** [comment]
Awesome, thanks.

**Comment by RockinChaos** [comment]
https://pastebin.com/vCgNQB8J

There ya go, tested and working ^~^

**Comment by chumpydo** [comment]
That was fast;

Let me test :)

**Comment by chumpydo** [comment]
Reloaded and doesn't seem to be working. I also tried `/itemjoin give book-item` and it didn't seem to work, claiming this message: http://i.imgur.com/kQMROSO.png

**Comment by RockinChaos** [comment]
Your using ItemJoin v4.1 correct?

**Comment by chumpydo** [comment]
Indeed.

**Comment by RockinChaos** [comment]
Somethings not right because Its working for me haha.

Can you send your config.yml/en-lang.yml and repost the items.yml that you are now using.

Also to double check your trying to get your items in the world named world right? You can check this by typing /itemjoin world
if it outputs - world then you are good to go.

**Comment by chumpydo** [comment]
Hang on - I'm in a world that is not named `world`, (it still outputs though), is that where I fucked up?

**Comment by RockinChaos** [comment]
Stand in the world you want to recieve the items in and type
/itemjoin world

If it doesnt say world then you defined it wrong.

enabled-worlds in the items.yml must be set to the name of the world you are trying to get the items in.

**Comment by chumpydo** [comment]
Not following you. Here's what it's outputting:
http://i.imgur.com/EX8ypxT.png

**Comment by RockinChaos** [comment]
Take the output that you just got. (the one scribbled in blue).

Go to your items.yml and find 
**enabled-worlds: world.**

change that to

enabled-worlds: nameofoutputtedworldhere

**Comment by chumpydo** [comment]
Will do. Sorry for scribbling it in blue, name gives away what we're working on.

Let me try and test and report back.

**Comment by RockinChaos** [comment]
Its all good.

**Comment by chumpydo** [comment]
Yeah! It worked! Alrighty, thanks a lot for your help.

Issue **resolved**, thread **closed**.

**Comment by RockinChaos** [comment]
Sweet, if you need any more help feel free to open another issue.

**Comment by chumpydo** [comment]
Also - is there a way to prevent the player from moving or dropping the item from their inventory?

**Comment by RockinChaos** [comment]
How do I set custom itemflags on an item?
Itemflags allow HUGE modifications to the items. All you have to do to add itemflags to an item, is add the itemflags identifier to the item as shown below.

These are custom itemflags, all itemflags with description are listed below, Itemflags are always being updated an added onto so check back here for updates!

Current available itemflags:
first-join - This will only give the item when joining the server for the first time, to get the item every time you join just remove this itemflag.
unbreakable - Set the item as INDESTRUCTIBLE!
hide-attributes - Hides ALL attributes, such as firework color, damage the item does, enchantments. (Hides everything from lore except the custom lore).
hide-durability - Hides the durability bar from the item.
placement - Prevents this item from being placed if it is a BLOCK.
inventory-modify - Prevents this item from being moved, stored, or switched to offhand in the players inventory.
cancel-events - Prevents player from opening a chest while this item is in hand or any open-able items. Useful for canceling the event of opening a book to set custom book items.
count-lock - Prevents the player from using all of the item. If the item is set to count: 32 then each time the player places the block item the stack will be refreshed back to 32.
death-drops - Clears this item if it is dropped on the ground when the player dies.
self-drops - Prevents the player from dropping this item, either clicking it out of inventory or using the drop key, usually (q).
disposable - If a command is ran from the item it will remove x1 of the item from the players inventory.
AllowOpBypass - This would allow the user IF THEIR OP to bypass most of these itemflags. It will work on ALL non item modifying flags. This would NOT work on unbreakable, hide-attributes, stuff like that.
CreativeBypass - This would allow the user IF THEIR IN CREATIVE MODE to bypass most of these itemflags. It will work on ALL non item modifying flags. This would NOT work on unbreakable, hide-attributes, stuff like that.

Here is an example of an item with all itemflags;
Code (Text):
items:
  ultra-item:
    id: DIAMOND_SWORD
    slot: 0
    itemflags: first-join, unbreakable, hide-attributes, hide-durability, placement, inventory-modify, cancel-events, count-lock, death-drops, self-drops, disposable, AllowOpBypass, CreativeBypass
    enabled-worlds: world


Basically simply add
itemflags: inventory-modify, self-drops, death-drops
